### PR TITLE
Attack chains fix

### PR DESCRIPTION
**Pull request by YiscahLevySilas1**

### CODE REVIEW

File: attackchains/attackchainhandler.go [hunk 1]
```diff
@@ -154,23 +154,41 @@ func (h *AttackChainsEngine) DetectAllAttackChainsFromLists(postureResourceSumma
...
-		attackChains = append(attackChains, currentAttackChains...)
+				attackChains = append(attackChains, currentAttackChains...)
+			}
+		}
 	}
 
 	return attackChains, nil
 }
 
+func (h *AttackChainsEngine) workLoadMatch(postureResourceSummary *armotypes.PostureResourceSummary, vul *cscanlib.CommonContainerScanSummaryResult) bool {
```
Comment (kooomix) [inline attackchains/attackchainhandler.go:178]: So you can remove the existing function "validateWorkLoadMatch"

File: attackchains/attackchainhandler.go [hunk 2]
```diff
@@ -154,23 +154,41 @@ func (h *AttackChainsEngine) DetectAllAttackChainsFromLists(postureResourceSumma
...
-		attackTracks, err := h.DetectAllAttackChains(postureResourceSummaries[i], vuls[i])
-		if err != nil {
-			return nil, err
-		}
+		for j := range vuls {
+			if h.workLoadMatch(postureResourceSummaries[i], vuls[j]) {
+				attackTracks, err := h.DetectAllAttackChains(postureResourceSummaries[i], vuls[j])
+				if err != nil {
+					return nil, err
+				}
```
Comment (kooomix) [inline attackchains/attackchainhandler.go:162]: Let's print an error if workLoad doesn't match.

**Review by kooomix — CHANGES_REQUESTED** [review]
If you are already making changes, maybe you can also fix the  GenerateAttackChainID function to generate the hash.

File: attackchains/attackchainhandler.go (hunk 1 above)
Comment (YiscahLevySilas1) [reply attackchains/attackchainhandler.go:178]: didn't realized we already had this func, I'll use it (I'll just call it earlier)

File: attackchains/attackchainhandler.go (hunk 2 above)
Comment (YiscahLevySilas1) [reply attackchains/attackchainhandler.go:162]: I don't think it's an error if they don't match, we're not expecting the vuln and postureResourceSummary to always be in the same index in their respective lists

**Review by kooomix — APPROVED** [review]

**Pull request merged**